src/game/Spell.cpp

```diff
@@ -5330,6 +5330,21 @@ SpellCastResult Spell::CheckCast(bool strict)
         }
     }
 
+    switch (m_spellInfo->Id)
```

Comment:
Possibly this part could be moved to separate "HandleCustomTargetChecking" function because there is a lot of it needed at least in TBC era (only 30 quests in and I already have 4 conditioned spells here)